### ISSUE TITLE
Easy local deployment

### ISSUE THREAD

**Issue by N0W0RK** [LSS-Manager/LSSM-V.4#2941]
### Hello world!

As ist currently stands, the build scripts are basically ready for easy local deployment. However the `LSSM_PORT` environment variable has to be set by hand or the server has to be started once. 

What would be the prefered way of modifying the build scripts to set the environment variable before build is executed? Can this step just be included with the full build or would that interfere with production deployment?

**Comment by Cr4zyc4k3** [comment]
Wdym with "the LSSM_PORT environment variable has to be set by hand or the server has to be started once."? If no `--port` was given the script will use [36551](https://github.com/LSS-Manager/LSSM-V.4/blob/a5347acd22d6fe1327b4984944efc466a51c4d1f/build/build.sh#L150). In case you want to change the default port in your dev enviroment you can do it there.

For local development you can run `./build/build.sh --local` which will run all required steps and spin up a web server with the set port. 

The `LSSM_PORT` variable is not used in production. We use a different web server (nginx) for that.

**Comment by N0W0RK** [comment]
When executing `yarn dev` to build for local deployment, the `LSSM_PORT` environment variable has to be set, so that `localhost` is set as the address for the userscript and other stuff in `src/config.ts`.

**Comment by Cr4zyc4k3** [comment]
Yeah, that happens when swapping the CD/CI-Pipelines but not adjusting the script :D
The `yarn dev` command was not meant to be used in that way. Back in the days it was used to create a build on private servers. So there is no need to set the `LSSM_Port` since this command won't create a local user script. 

If you just want to test your docs you can run `./build/build.sh --docs` followed by `ws -d ./dist/docs/` or ` php -S localhost:8000 -t ./dist/docs/` or `python -m http.server -d ./dist/docs` or a similiar web server.

**Comment by N0W0RK** [comment]
I am currently working on #2951 . When this is finished, I might take a look at the build scripts to ease local dev a little bit

**Comment by Cr4zyc4k3** [comment]
As mentioned above: `./build/build.sh --local` will produce a local minimal working version but without docs. Looking forward to see a cleaned up building process :smile:
